app/services/api/v1x2/catalog/evaluate_order_process.rb

```diff
@@ -0,0 +1,68 @@
+module Api
+  module V1x2
+    module Catalog
+      class EvaluateOrderProcess
+        def initialize(order)
+          @order = order
+        end
+
+        def process
+          applicable_order_item = @order.order_items.first
+
+          relevant_order_processes = find_relevant_order_processes
+
+          # TODO: Put the order processes in order by their sequence number before
+          # doing this logic
+          @before_sequence_number = 1
+          @after_sequence_number = (relevant_order_processes.length * 2) + 1
+
+          relevant_order_processes.each do |order_process|
+            Catalog::AddToOrderViaOrderProcess.new(before_params(order_process)).process
+            Catalog::AddToOrderViaOrderProcess.new(after_params(order_process)).process
+
+            @before_sequence_number += 1
+            @after_sequence_number -= 1
+          end
+
+          applicable_sequence = (relevant_order_processes.length + 1)
+          applicable_order_item.update(:process_sequence => applicable_sequence, :process_scope => "applicable")
+
+          self
+        end
+
+        private
+
+        def find_relevant_order_processes
+          order_process_ids = TagLink.where(:tag_name => all_tags).pluck(:order_process_id)
+
+          OrderProcess.where(:id => order_process_ids)
```

Comment:
Try not to resolve actual ids by `pluck`. In stead try to come up with a select statement and pass it to `where` here. Together it is one SQL query. Another benefit is that it avoids writing a long list of ids in the log. Look at the example from https://github.com/RedHatInsights/approval-api/blob/master/app/services/workflow_find_service.rb#L25

Reply:
To preface my comment: I agree, I can change it, but I'm curious why then https://github.com/RedHatInsights/approval-api/blob/master/app/services/workflow_find_service.rb#L5, which is in the very same file that you mentioned, is going against that pattern, as well as https://github.com/RedHatInsights/catalog-api/blob/master/app/services/api/v1x2/catalog/get_linked_order_process.rb#L9.

I'm not trying to argue that "well it's done like this here in these other places, so why isn't this one OK too?", but I'm just wondering if this is a broader thing that we should be looking out for more often and those ones just happened to slip through, or if it's really not as big of a problem.

Reply:
Yes. We should refactor the places you mentioned.